### PR TITLE
add pre-down hooks to remove dataplane resource and unblock azd down

### PR DESCRIPTION
**Pull request by hund030**
## Purpose

Fix azd down issue in https://github.com/Azure-Samples/openai-chat-vision-quickstart/issues/5

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
* Run azd up
* Run azd down

### CODE REVIEW

**Comment by anfibiacreativa** [comment]
This can be an interim solution if accepted by template owners, but we should continue to investigate an azd-based solution without hooking scripts with az

File: scripts/pre-down.ps1 [hunk 1]
```diff
@@ -0,0 +1 @@
+az cognitiveservices account deployment delete --name $env:AZURE_OPENAI_RESOURCE_NAME --resource-group $env:AZURE_RESOURCE_GROUP --deployment-name $env:AZURE_OPENAI_CHATGPT_DEPLOYMENT
```
Comment (pamelafox) [inline scripts/pre-down.ps1:1]: Wouldn't this need to use "azd env get-value AZURE_OPENAI_RESOURCE_NAME"? It's not necessarily in the global env- I've been trying to put as little in global env as possible lately, and fetching directly from azd when needed.

File: azure.yaml [hunk 2]
```diff
@@ -7,4 +7,14 @@ services:
...
+hooks:
+  predown:
+    windows:
+      shell: pwsh
+      run: ./scripts/pre-down.ps1
+      continueOnError: false
+    posix:
+      shell: sh
+      run: ./scripts/pre-down.sh
+      continueOnError: false
```
Comment (pamelafox) [inline azure.yaml:20]:
```suggestion
      continueOnError: false

```

File: scripts/pre-down.ps1 (hunk 1 above)
Comment (hund030) [reply scripts/pre-down.ps1]: Thanks for the suggestion. Updated.

**Comment by hund030** [comment]
> This can be an interim solution if accepted by template owners, but we should continue to investigate an azd-based solution without hooking scripts with az

I agree that it's better not to introduce az cli dependency. I try to replace it with Python SDK. Does it look better to you?

This is still in testing, please do not merge it before I done.

**Review by pamelafox — APPROVED** [review]
I am fine with this, assuming it works!

**Pull request merged**